geopandas/tools/sjoin.py

```diff
@@ -79,11 +89,11 @@ def sjoin(
     if right_df._sindex_generated or (
         not left_df._sindex_generated and right_df.shape[0] > left_df.shape[0]
     ):
-        tree_idx = right_df.sindex
-        tree_idx_right = True
+        tree_idx_r = right_df.sindex
+        tree_idx_l = None
     else:
-        tree_idx = left_df.sindex
-        tree_idx_right = False
+        tree_idx_r = None
+        tree_idx_l = left_df.sindex
```

Comment:
These flags don't _need_ to be changed, I just found it more convenient to pass the parameters this way. Now whichever index is not `None` is the one to use.